File: server/src/scraper/entities/sonata_effect.entity.ts
import {Column, Entity, JoinTable, ManyToMany, OneToMany, PrimaryGeneratedColumn, Unique} from "typeorm";
import {EchoEntity} from "./echo.entity";

@Entity()
@Unique(['name', 'index'])
export class SonataEffectEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    name: string

    @Column()
    imageUrl: string

    @Column()
    index: number

    @Column({nullable: true})
    two_set: string

    @Column({nullable: true})
    five_set: string

    @Column({nullable: true})
    color: string

    @ManyToMany(() => EchoEntity, echo => echo.sonataEffects)
    @JoinTable()
    echoes: EchoEntity[];
}

File: mobile/lib/domains/sonata_effect/sonata_effect_entity.dart


import 'package:freezed_annotation/freezed_annotation.dart';

part 'sonata_effect_entity.freezed.dart';
part 'sonata_effect_entity.g.dart';
@freezed
class SonataEffectEntity with _$SonataEffectEntity {
  const factory SonataEffectEntity({
    int? id,
    String? name,
    String? imageUrl,
    int? index,
    String? color,
    String? two_set,
    String? five_set,
  }) = _SonataEffectEntity;

  factory SonataEffectEntity.fromJson(Map<String, dynamic> json) =>
      _$SonataEffectEntityFromJson(json);
}
